### PR TITLE
Adds `CustomTextField` component

### PR DESCRIPTION
**Pull request by ggirotto**

### CODE REVIEW

File: test/application/widgets/theme/custom_text_field_test.dart [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:memo/application/theme/theme_controller.dart';
+import 'package:memo/application/widgets/theme/custom_text_field.dart';
+
+import '../../../utils/widget_pump.dart';
+
+void main() {
+  group('Editing State - ', () {
+    testWidgets('should have the expected text in the controller after typing in the TextField', (tester) async {
```
Comment (matuella) [inline test/application/widgets/theme/custom_text_field_test.dart:10]:
```suggestion
    testWidgets('should have the expected text in the controller after typing', (tester) async {
```

File: test/application/widgets/theme/custom_text_field_test.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      final textField = CustomTextField(controller: controller);
+      const expectedText = 'any';
+
+      await pumpMaterialScopedWithTheme(tester, textField);
+      await tester.enterText(find.byType(CustomTextField), expectedText);
+
+      expect(controller.text, expectedText);
+    });
+
+    testWidgets('should have focus when tapping in the TextField', (tester) async {
```
Comment (matuella) [inline test/application/widgets/theme/custom_text_field_test.dart:21]:
```suggestion
    testWidgets('should have focus when being tapped', (tester) async {
```

File: lib/application/widgets/theme/custom_text_field.dart [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        highlightColor: Colors.transparent,
+        icon: Image.asset(images.clearAsset, color: neutralSwatch.shade400),
+        onPressed: () {
+          onTap();
+          onChanged?.call(null);
+        },
+      );
+    } else {
+      return suffixIcon;
+    }
```
Comment (matuella) [inline lib/application/widgets/theme/custom_text_field.dart:174]: Are all of these scenarios tested?

File: test/application/widgets/theme/custom_text_field_test.dart [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      expect(renderedTextField.decoration!.suffixIcon, null);
+
+      await tester.enterText(find.byType(CustomTextField), 'any');
+      await tester.pumpAndSettle();
+
+      renderedTextField = find.byType(TextField).first.evaluate().single.widget as TextField;
+      expect(renderedTextField.decoration!.suffixIcon, isA<IconButton>());
+    });
+
+    testWidgets('should show custom suffix icon when not focused', (tester) async {
```
Comment (matuella) [inline test/application/widgets/theme/custom_text_field_test.dart:47]: Maybe missing test that should show when showsClearIcon is false or has text?

File: test/application/widgets/theme/custom_text_field_test.dart [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      final suffixIcon = Container(color: Colors.red);
+      final textField = CustomTextField(suffixIcon: suffixIcon);
+
+      await pumpMaterialScopedWithTheme(tester, textField);
+
+      final renderedTextField = find.byType(TextField).first.evaluate().single.widget as TextField;
+      expect(renderedTextField.decoration!.suffixIcon, suffixIcon);
+    });
+
+    testWidgets('should replace custom suffix icon by clear button when editing', (tester) async {
```
Comment (matuella) [inline test/application/widgets/theme/custom_text_field_test.dart:57]: What about this
```suggestion
    testWidgets('should show clear button when has text, is focused and showsClearIcon is true', (tester) async {
```

File: test/application/widgets/theme/custom_text_field_test.dart [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+      await tester.enterText(find.byType(CustomTextField), 'any');
+      await tester.pumpAndSettle();
+
+      renderedTextField = find.byType(TextField).first.evaluate().single.widget as TextField;
+      expect(renderedTextField.decoration!.suffixIcon, isA<IconButton>());
+    });
+  });
+
+  group('Error State - ', () {
```
Comment (matuella) [inline test/application/widgets/theme/custom_text_field_test.dart:73]: "Labels"

What about testing helperText as well here, and while at it, maybe test if both are supplied and who takes precedence?

**Review by matuella — CHANGES_REQUESTED** [review]

File: test/application/widgets/theme/custom_text_field_test.dart (hunk 6 above)
Comment (ggirotto) [reply test/application/widgets/theme/custom_text_field_test.dart:73]: Awesome suggestion

File: lib/application/widgets/theme/custom_text_field.dart (hunk 3 above)
Comment (ggirotto) [reply lib/application/widgets/theme/custom_text_field.dart:174]: Added more use-cases

**Review by matuella — APPROVED** [review]

**Pull request merged**